### ISSUE TITLE
Update frequency question

### ISSUE THREAD

**Issue by hmwelsch** [robertsLando/node-red-contrib-m-bus#21]
Hi robertsLando,
thank you very much for this very nice peace of work. I am a new user of your contribution.

Is there any way to influence the update frequency of the M-Bus devices? I would really only need to get daily updates and would like to avoid the overhead and system load with more frequent updates ...

Since this is not really an issue, please feel free to dismiss/delete this entry. If it is inappropriate to ask questions this way, please let me know how to do better. I have tried to somehow find another way, but unfortunately without success.

Best regards,
hmw

**Comment by robertsLando** [comment]
Actually the mbus-out node outputs devices data as fast as he can read them in series, but you could also trigger reads manually by using mbus-controller and `getData` action with the required address to read, you could use a trigger node that triggers that command daily to read them daily, BTW I suggest to simply keep last data stored in an object and simply read the values from there daily as reads can take long with mbus

**Comment by hmwelsch** [comment]
Excellent! Will do. Thank you very much for your prompt and helpful reply! Have a nice day.

**Comment by hmwelsch** [comment]
Perfect response. Issue solved.

**Comment by hmwelsch** [comment]
Hi,

thanks again. I have another remark and another question regarding the function setPrimary.

The first remark concerns a little typo in the bus-controller.js line 121:
                client.emit('mbCommandDone', 'Setted new primary ID ' + cmd.new + ' to device ' + cmd.old);
The correct English past tense for Set is Set instead of Setted ;-) So is should actually read:
                client.emit('mbCommandDone', 'Set new primary ID ' + cmd.new + ' to device ' + cmd.old);
Or better readable probably:
                client.emit('mbCommandDone', ‚New primary ID ' + cmd.new + ‚ set to device ' + cmd.old);
This is of course nothing of any importance ...

The question though is more important for me:
I have two M-Bus masters at work (2 stacked M-Bus HATs for Raspberry Pi). Both work perfectly well for scanning and readDevice(s). However, on one of them I could successfully use setPrimary, while on the other one I get „Error while setting new primary ID 100 of device 30101867523B0102“.

How can I find out more about the root cause of this error?

Best regards
Martin Welsch

www.mwitplus.com <http://www.mwitplus.com/> • IT Trends • Technology and more • Tel. +49 7032 508634

**Comment by robertsLando** [comment]
> How can I find out more about the root cause of this error?

You could try to check node-red logs for that, if installed on rpi the command to tail logs should be `node-red-logs`

**Comment by robertsLando** [comment]
Let me release a new version with bumped deps and better error logs

**Comment by hmwelsch** [comment]
Sure. I’ll be waiting for that. Thanks.

**Comment by robertsLando** [comment]
Try using version 3.0.0-beta.0:

```
npm i node-red-contrib-m-bus@3.0.0-beta.0
```

**Comment by hmwelsch** [comment]
Thanks. That was it. The response was this time a bit more explanatory. It turned out, that the primary IDs were all set. Thus the attempt to do it (again) resulted in this error message. However, since I did not see the filed Primary Address in the response, I did not know that it was already there. Now the response from getDevice(s) also shows all correct primaryIds. So it seems I am all set now.

When I receive the next update, will I automatically return to the standard path or should I manually do something to ensure this?

**Comment by robertsLando** [comment]
> When I receive the next update, will I automatically return to the standard path or should I manually do something to ensure this?

What do you mean? 

If your goal is to read each device once a day what I would do is to keep auto scan enabled in client setting and just store the values you need in the flow context using `flow.set('deviceaddresshere', { valueX: X, valueY:Y })`. Then once a day at a specific time simply read those value from flow using flow.get and send them where you have to.

**Comment by hmwelsch** [comment]
Sorry, I was not clear in my response. I have no open questions or issues. I was only asking if it makes any difference that I have now upgraded to your beta version. I would like to stay in the „mainstream“ path in future and not miss further advances. Von meinem iPhone gesendetAm 09.11.2022 um 19:11 schrieb Daniel Lando ***@***.***>:﻿

When I receive the next update, will I automatically return to the standard path or should I manually do something to ensure this?

What do you mean?
If your goal is to read each device once a day what I would do is to keep auto scan enabled in client setting and just store the values you need in the flow context using flow.set('deviceaddresshere', { valueX: X, valueY:Y }). Then once a day at a specific time simply read those value from flow using flow.get and send them where you have to.

**Comment by robertsLando** [comment]
Oh now I get what you mean, sorry. I released a beta because I don't have a test device with me, I bumped deps and fixed all the code without actually testing it. 

If you report that it's working without any issue I will release it as stable now. Ok?

PS: Thanks for the contribution, much appreciated 🙏🏼

**Comment by hmwelsch** [comment]
Yes. Everything seems to work well. No problems with your beta so far. Should be OK to merge. Von meinem iPhone gesendetAm 10.11.2022 um 08:23 schrieb Daniel Lando ***@***.***>:﻿
Oh now I get what you mean, sorry. I released a beta because I don't have a test device with me, I bumped deps and fixed all the code without actually testing it.
If you report that it's working without any issue I will release it as stable now. Ok?
PS: Thanks for the contribution, much appreciated 🙏🏼
